src/include/migraphx/filesystem.hpp

```diff
@@ -64,6 +64,18 @@
 #error "No filesystem include available"
 #endif
 
+// NOLINTBEGIN
+#ifdef _WIN32
+#define MIGRAPHX_EXE_POSTFIX ".exe"
+#define MIGRAPHX_LIB_PREFIX ""
+#define MIGRAPHX_LIB_POSTFIX ".dll"
+#else
+#define MIGRAPHX_EXE_POSTFIX ""
+#define MIGRAPHX_LIB_PREFIX "lib"
+#define MIGRAPHX_LIB_POSTFIX ".so"
```

Comment:
First, this shouldn't go in this header(which is just used to include std::filesystem or the filesystem TS). Secondly, we shouldn't use macros for this. We should have functions that can append the extension to the name.